api/nimbus_backend.go

```diff
@@ -0,0 +1,1027 @@
+	"github.com/status-im/status-go/appdatabase"
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/logutils"
+	"github.com/status-im/status-go/multiaccounts"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/node"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/rpc"
+	accountssvc "github.com/status-im/status-go/services/accounts"
+	nimbussvc "github.com/status-im/status-go/services/nimbus"
+	"github.com/status-im/status-go/services/personal"
+	"github.com/status-im/status-go/services/rpcfilters"
+	"github.com/status-im/status-go/services/subscriptions"
+	"github.com/status-im/status-go/services/typeddata"
+	"github.com/status-im/status-go/signal"
+	"github.com/status-im/status-go/transactions"
+)
+
+// const (
+// 	contractQueryTimeout = 1000 * time.Millisecond
+// )
+
+var (
+	// ErrWhisperClearIdentitiesFailure clearing whisper identities has failed.
+	ErrWhisperClearIdentitiesFailure = errors.New("failed to clear whisper identities")
+	// ErrWhisperIdentityInjectionFailure injecting whisper identities has failed.
+	ErrWhisperIdentityInjectionFailure = errors.New("failed to inject identity into Whisper")
+	// ErrUnsupportedRPCMethod is for methods not supported by the RPC interface
+	ErrUnsupportedRPCMethod = errors.New("method is unsupported by RPC interface")
+	// ErrRPCClientUnavailable is returned if an RPC client can't be retrieved.
+	// This is a normal situation when a node is stopped.
+	ErrRPCClientUnavailable = errors.New("JSON-RPC client is unavailable")
+)
+
+var _ StatusBackend = (*nimbusStatusBackend)(nil)
+
+// nimbusStatusBackend implements the Status.im service over Nimbus
+type nimbusStatusBackend struct {
+	StatusBackend
+
+	mu sync.Mutex
+	// rootDataDir is the same for all networks.
+	rootDataDir string
+	appDB       *sql.DB
+	statusNode  *node.NimbusStatusNode
+	// personalAPI     *personal.PublicAPI
+	// rpcFilters      *rpcfilters.Service
+	multiaccountsDB *multiaccounts.Database
+	accountManager  *account.GethManager
+	// transactor      *transactions.Transactor
+	connectionState         connectionState
+	appState                appState
+	selectedAccountShhKeyID string
+	log                     log.Logger
+	allowAllRPC             bool // used only for tests, disables api method restrictions
+}
+
+// NewNimbusStatusBackend create a new nimbusStatusBackend instance
+func NewNimbusStatusBackend() *nimbusStatusBackend {
+	defer log.Info("Status backend initialized", "backend", "nimbus", "version", params.Version, "commit", params.GitCommit)
+
+	statusNode := node.NewNimbus()
+	accountManager := account.NewGethManager()
+	// transactor := transactions.NewTransactor()
+	// personalAPI := personal.NewAPI()
+	// rpcFilters := rpcfilters.New(statusNode)
+	return &nimbusStatusBackend{
+		statusNode:     statusNode,
+		accountManager: accountManager,
```

Comment:
Same here. All this code duplication just because we want to use a different `AccountManager`? Is that really wise?